master/internal/api_experiment.go

```diff
@@ -390,6 +390,20 @@ func (a *apiServer) GetExperiments(
 		query = query.Where("e.owner_id IN (?)", bun.In(req.UserIds))
 	}
 
+	if req.ExperimentFilter != nil {
+		expRestriction := req.ExperimentFilter.Restriction.String()
+		experimentIds := make([]int32, 0)
+		if req.ExperimentFilter.ExperimentIds != nil {
+			experimentIds = append(experimentIds, req.ExperimentFilter.ExperimentIds...)
+		}
+		if expRestriction == apiv1.GetExperimentsRequest_ExperimentFilter_Restriction_name[1] {
+			query = query.Where("e.id = ANY (ARRAY[?]::integer[])", bun.In(experimentIds))
+		} else if expRestriction == apiv1.GetExperimentsRequest_ExperimentFilter_Restriction_name[2] &&
+			len(experimentIds) > 0 {
+			query = query.Where("NOT (e.id = ANY (ARRAY[?]::integer[]))", bun.In(experimentIds))
```

Comment:
```suggestion
		if expRestriction == apiv1.GetExperimentsRequest_ExperimentFilter_Restriction_name[1] {
			query = query.Where("e.id IN (?)", bun.In(experimentIds))
		} else if expRestriction == apiv1.GetExperimentsRequest_ExperimentFilter_Restriction_name[2] &&
			len(experimentIds) > 0 {
			query = query.Where("e.id  NOT IN (?)", bun.In(experimentIds))
```

Reply:
I use `ANY (ARRAY[?]::integer[]` to handle empty list just in case. Or do i need to handle it in Go instead of SQL?

Reply:
perhaps skip doing `where` if it's empty?

Reply:
ok I'll change Go code instead of SQL then

Reply:
note:
I just remember why I implemented this way. If there's no item in array with `INCLUDE`, return 0 experience.  If there's no item in array with `EXCLUDE`, return all experiences.
That was my intention. 
I'll change the code with `IN` but its just a note

Reply:
is there a valid use case for having no items in include list? maybe that should just throw an error, or return an empty response early?
constructing and running a query for that seems unnecessary.

Reply:
I can't come up with any valid use case but I just thought it's intuitive. 
> maybe that should just throw an error, or return an empty response early?

I'll go with this way

Reply:
>I just remember why I implemented this way. If there's no item in array with INCLUDE, return 0 experience. If there's no item in array with EXCLUDE, return all experiences.
That was my intention. IN doesnt accept empty array, that's why I used it
I'll change the code with IN but its just a note

Just remembered (again) that why I wanted it to be behaved like this.
It's because some projects could be empty or there could be no pinned item in a project. In these cases, this behavior is useful. 
I rewrote the code to archive the behavior with `IN` instead of using `ANY` clause, but the code became a bit uglier